scripts/review_comments_retriver.py

```diff
@@ -0,0 +1,205 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+
+import os
+import re
+from collections import defaultdict
+from functools import lru_cache
+
+import tenacity
+from libmozdata.phabricator import PhabricatorAPI
+from tqdm import tqdm
+from unidiff import Hunk, PatchedFile, PatchSet, UnidiffParseError
+
+from bugbug import db, phabricator
+from bugbug.tools.code_review import InlineComment
+from bugbug.utils import get_secret
+from bugbug.vectordb import QdrantVectorDB, ReviewCommentsDB
+
+
+class PhabricatorReviewCommentRetriever:
+    NIT_PATTERN = re.compile(r"[^a-zA-Z0-9]nit[\s:,]", re.IGNORECASE)
+
+    def __init__(self) -> None:
+        os.makedirs("patches", exist_ok=True)
+        self.phabricator_api = PhabricatorAPI(
+            get_secret("PHABRICATOR_TOKEN"),
+            get_secret("PHABRICATOR_URL"),
+        )
+
+        db.download(phabricator.REVISIONS_DB)
+
+    @lru_cache(maxsize=1)
+    @tenacity.retry(
+        stop=tenacity.stop_after_attempt(7),
+        wait=tenacity.wait_exponential(multiplier=1, min=16, max=64),
+        reraise=True,
+    )
+    def load_patch_set(self, diff_id):
+        try:
+            with open(f"patches/{diff_id}.patch", "r") as f:
+                patch = f.read()
+        except FileNotFoundError:
+            with open(f"patches/{diff_id}.patch", "w") as f:
+                patch = self.phabricator_api.load_raw_diff(diff_id)
```

Comment:
We could do like https://github.com/mozilla/bugbug/blob/7722d95d124e5bf01bb9b7eb65825773ca29b9df/scripts/code_review_tool_runner.py#L21 to make it more generic

Reply:
Done as part of moving generic methods into `ReviewData`: 058666bd56310b95b8d4064b685ffe5c91fb48c7.